### ISSUE TITLE
missing way to get RowVersion when doing SQL based update?

### ISSUE THREAD

**Issue by fantasyz** [oracle/nosql-dotnet-sdk#21]
Hi,

The situation is a little awkward. To ensure any update is not overwriting newer data on the DB, we can provide a **RowVersion** along with the Put call to the DB. Everything is good until we use SQL-based method to update the records. We don't get the updated RowVersion when executing SQL. It also doesn't seems like efficient to do another primary key Get call just to retrieve the new RowVersion. Thus, I would like to suggest the SQL operation to also return a RowVesion for the record being updated. 

(according to the Doc, the current implementation only allow max of 1 row to be updated by SQL)

**Comment by dario-vega** [comment]
Table rows are record values conforming to the table schema but with some additional properties that are not part of the table schema. To extract the value of such properties, the functions listed in the documentation could be used (link [here](https://docs.oracle.com/en/database/other-databases/nosql-database/22.3/sqlreferencefornosql/functions-rows.html)). You can use the `row_version` function, it returns a binary value.  I am reviewing why the `row_version` function is not currently documented. 

```
UPDATE blogtable $t 
set blog = "blog missing way to get RowVersion"
WHERE id = 1  
RETURNING $t, 
cast(row_version($t) as String) as rowVersionAsString, 
row_version($t) as rowVersion
```

you can even create a JSON doc instead of returning columns

```
UPDATE blogtable $t 
set blog = "blog missing way to get RowVersion"
WHERE id = 1  
RETURNING  { "JSONdoc":$t, 
   "metadata": {
      "rowversionAsString" : cast(row_version($t) as String), 
      "rowversionAsBinary": row_version($t),
      "modification_time" : modification_time($t) 
  }
} as row

```

You can also use this function in the WHERE clause

```
UPDATE blogtable $t 
set blog = "blog missing way to get RowVersion"
WHERE id = 1 
AND row_version($t) = cast ("rO0ABXcsABtr+qkrdXtNTY0pyeZgVd9PAAAAAAA9ipEBAwAAAA8AAAACAAAAFQv6a5k=" as binary)
RETURNING $t, 
cast(row_version($t) as String) as rowVersionAsString, 
row_version($t) as rowVersion
```


What is your use-case to provide better insights? Let us know if it is what you are looking for

**Comment by fantasyz** [comment]
Hi, 

I just tried the row_version() and I can successfully get the version. However, I cannot follow your last example. 

The problem is that RowVersion class didn't provide any way to access the value at all. Not even the ToString() will return the value. I don't have the value to pass to PreparedStatement.SetVariable(). In fact, I am currently holding the RowVersion instance for Put function. Please expose the properties of this class. Preferably allow us to create a new one so that we can just hold the string value from our own classes and convert it back to RowVersion type when needed.

The current RowVersion (from DLL non-editable):

```C#
public class RowVersion
{
	internal byte[] Bytes { get; }

	internal string Encoded
	{
		get
		{
			if (Bytes == null)
			{
				return null;
			}

			return Convert.ToBase64String(Bytes);
		}
	}

	internal RowVersion(byte[] data)
	{
		Bytes = data;
	}
}
```

Expected:

```C#
public class RowVersion
{
	public byte[] Bytes { get; set; } = null;
	public string Encoded => Bytes == null ? null : Convert.ToBase64String(Bytes);

	public RowVersion(byte[] data)
	{
		Bytes = data;
	}

	public override string ToString() => Encoded;
}
```

**Comment by gmfeinberg** [comment]
@y-polonsk can you please look into this? The other NoSQL language SDKs allow creation of the version object from a byte array, which is necessary for this use case (using a version acquired via query in the API)

**Comment by y-polonsk** [comment]
Ok, thanks, will fix.

**Comment by dario-vega** [comment]
@fantasyz the PR #22 fix your issue. Can you provide us with your current status - the impact of it?

**Comment by fantasyz** [comment]
Thanks for the update. I was just testing the capability of the Oracle NoSql on a new project. I think this issue can be closed now.

**Comment by dario-vega** [comment]
It will be included in the next release soon. 

Do not hesitate to contact us if you have questions. Use https://github.com/oracle/nosql-examples/discussions/categories/q-a

**Comment by dario-vega** [comment]
fixed by PR #22
